### ISSUE TITLE
Validation broken on azurerm_api_management_named_value.display_name

### ISSUE THREAD

**Issue by antgustech** [hashicorp/terraform-provider-azurerm#24856]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.7.3

### AzureRM Provider Version

3.91.0

### Affected Resource(s)/Data Source(s)

3.91.0

### Terraform Configuration Files

```hcl
resource "azurerm_api_management_named_value" "my_hostname" {
  display_name        = "my_hostname"
}
```

### Debug Output/Panic Output

```shell
│ Error: "display_name" may only contain alphanumeric characters, periods, underscores and dashes
```

### Expected Behaviour

Would expect underscores to work.

### Actual Behaviour

 Error during plan

### Important Factoids

Reverting to azurerm version 3.87.0 seems to resolve the issue.

**Comment by antgustech** [comment]
![Screenshot 2024-02-12 141130](https://github.com/hashicorp/terraform-provider-azurerm/assets/8721113/5ccc496d-aa9e-46e7-8a5d-4a3e39b98a33)
 
The portal UI says only dashes are supported now.

**Comment by kWozniak-tt** [comment]
We have the same issue after moving from hashicorp/azurerm v3.90.0
So there is regression between hashicorp/azurerm v3.90.0->v3.91.0

**Comment by clarkminor** [comment]
Also hitting this. Here are the debug logs:

```
2024-02-12T14:42:05.799-0600 [DEBUG] provider.terraform-provider-azurerm_v3.91.0_x5: AzureRM Request: 
	GET /subscriptions/SUBSCRIPTION_ID/resourceGroups/RESOURCE_GROUP/providers/Microsoft.ApiManagement/service/APIM_SERVICE/namedValues/backend-eastus2-1?api-version=2022-08-01 HTTP/1.1
	Host: management.azure.com
	User-Agent: HashiCorp/go-azure-sdk (Go-http-Client/1.1 namedvalue/2022-08-01) HashiCorp Terraform/1.5.2 (+https://www.terraform.io) Terraform Plugin SDK/2.10.1 terraform-provider-azurerm/3.91.0 pid-222c6c49-1b0a-5959-a213-6608f9eb8820
	Content-Type: application/json; charset=utf-8
	X-Ms-Correlation-Request-Id: 8f5d5c33-1da7-805c-f91c-3983a618e726
	Accept-Encoding: gzip: timestamp=2024-02-12T14:42:05.799-0600

2024-02-12T14:42:05.799-0600 [DEBUG] provider.terraform-provider-azurerm_v3.91.0_x5: GET https://management.azure.com/subscriptions/SUBSCRIPTION_ID/resourceGroups/RESOURCE_GROUP/providers/Microsoft.ApiManagement/service/APIM_SERVICE/namedValues/backend-eastus2-1?api-version=2022-08-01: timestamp=2024-02-12T14:42:05.799-0600

2024-02-12T14:42:06.157-0600 [DEBUG] provider.terraform-provider-azurerm_v3.91.0_x5: AzureRM Response for https://management.azure.com/subscriptions/SUBSCRIPTION_ID/resourceGroups/RESOURCE_GROUP/providers/Microsoft.ApiManagement/service/APIM_SERVICE/namedValues/backend-eastus2-1?api-version=2022-08-01: 
	HTTP/2.0 200 OK
	Content-Length: 717
	Cache-Control: no-cache
	Content-Type: application/json; charset=utf-8
	Date: Mon, 12 Feb 2024 20:42:05 GMT
	Etag: "AAAAAAACJbk="
	Expires: -1
	Pragma: no-cache
	Strict-Transport-Security: max-age=31536000; includeSubDomains
	X-Cache: CONFIG_NOCACHE
	X-Content-Type-Options: nosniff
	X-Ms-Correlation-Request-Id: 8f5d5c33-1da7-805c-f91c-3983a618e726
	X-Ms-Ratelimit-Remaining-Subscription-Reads: 11996
	X-Ms-Request-Id: 8f5d5c33-1da7-805c-f91c-3983a618e726
	X-Ms-Routing-Request-Id: EASTUS2:20240212T204206Z:41871143-5834-4669-a47f-c214d71b6624
	X-Msedge-Ref: Ref A: 7EA15F101DF542AF87DEA47FD5643F79 Ref B: MNZ221060610051 Ref C: 2024-02-12T20:42:06Z

	{
	  "id": "/subscriptions/SUBSCRIPTION_ID/resourceGroups/RESOURCE_GROUP/providers/Microsoft.ApiManagement/service/APIM_SERVICE/namedValues/backend-eastus2-1",
	  "type": "Microsoft.ApiManagement/service/namedValues",
	  "name": "backend-eastus2-1",
	  "properties": {
	    "displayName": "backend-eastus2-1",
	    "keyVault": {
	      "secretIdentifier": "https://KEYVAULT_NAME.vault.azure.net/secrets/backend-eastus2-1-primary-access-key/SECRET_ID",
	      "identityClientId": null,
	      "lastStatus": {
	        "code": "Success",
	        "timeStampUtc": "2024-02-12T18:48:08.2879659Z"
	      }
	    },
	    "tags": null,
	    "secret": true
	  }
	}: timestamp=2024-02-12T14:42:06.157-0600

2024-02-12T14:42:06.158-0600 [ERROR] provider.terraform-provider-azurerm_v3.91.0_x5: Response contains error diagnostic: diagnostic_severity=ERROR tf_proto_version=5.4 tf_provider_addr=provider tf_resource_type=azurerm_api_management_named_value tf_rpc=ValidateResourceTypeConfig diagnostic_summary=""display_name" may only contain alphanumeric characters, periods, underscores and dashes" @module=sdk.proto diagnostic_detail= @caller=github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/internal/diag/diagnostics.go:58 tf_req_id=07e28b3f-53da-e693-ec11-c3bfff34f774 diagnostic_attribute=AttributeName("display_name") timestamp=2024-02-12T14:42:06.158-0600
2024-02-12T14:42:06.158-0600 [ERROR] vertex "azurerm_api_management_named_value.backend-eastus2-1" error: ""display_name" may only contain alphanumeric characters, periods, underscores and dashes"
```

Shows that the AzureRM API returns `200` successfully and that our module is setting the display name to `backend-eastus2-1`. However, the validation still fails with `"display_name" may only contain alphanumeric characters, periods, underscores and dashes"`.

**Comment by arjundevane** [comment]
Introduced in https://github.com/hashicorp/terraform-provider-azurerm/pull/24749, and release 3.91.0.

Since this is potentially breaking change, should there be an opt-in mechanism for existing code? With a default of `false`.

**Comment by Felix-Franz** [comment]
Duplicate of #24831, was fixed and merged to main with #24832.

**Comment by arjundevane** [comment]
> Duplicate of #24831, was fixed and merged to main with #24832.

I think https://github.com/hashicorp/terraform-provider-azurerm/pull/24832 was just for allowing up to 265 characters. However this issue seems to be with some special characters in the name like underscore e.g. `my_hostname`.

**Comment by rcskosir** [comment]
@antgustech Thanks for taking the time to open this issue. @Felix-Franz Thank you for helping find the duplicate as well as the PR that has been merged to fix this issue. As such, I am going to mark this issue as closed with #24832 which should go out with v3.92.
